### ISSUE TITLE
anyopen should respect binary/text arguments

### ISSUE THREAD

**Issue by richardjgowers** [MDAnalysis/mdanalysis#2130]
`MDAnalysis.lib.util.anyopen` (and its context manager cousin `openany`) currently opens files/archives/streams seamlessly and is used in most Readers.  It accepts a mode argument like the regular `open` function, [but this mode doesn't explicitly support more complex modes](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/util.py#L348) ie `r+`.

In PR #2127 we started to require using `mode='rb'` to get ascii reading working in Windows environments.  This works currently, but isn't technically supported, ie the api for `anyopen` doesn't say this should work (and it's therefore not tested to work either).

These more complicated modes should be accepted & enforced (or at least a subset which we need to use, ie `'rb'` for backends (ie files/archives/streams) which support this).

**Comment by orbeckst** [comment]
See code for [anyopen()](https://github.com/MDAnalysis/mdanalysis/blob/89b2299e3ae8ab548fcc932b72eb6128b2be0463/package/MDAnalysis/lib/util.py#L333-L424).

**Comment by orbeckst** [comment]
We set the default `mode="rt"` https://github.com/MDAnalysis/mdanalysis/blob/89b2299e3ae8ab548fcc932b72eb6128b2be0463/package/MDAnalysis/lib/util.py#L333 and then say that we ignore everything beyond the first letter – that's rather inconsistent :-p.

**Comment by tylerjereddy** [comment]
Yes, should be clarified / consistent / tested, but useful that it is actually apparently possible for Windows of course.

**Comment by jbarnoud** [comment]
We also explicitly use 'rb' for the TRZ reader. I think I remember the 'rt' default was set because of python3 because some compressed files were open in byte mode. It starts to be an old change so I can be wrong.

**Comment by orbeckst** [comment]
Can we define clearly what needs to happen? Something like

- allow 'rb' and 'wb' (and 'rt' and 'wt') in addition to 'r', 'w', 'a'.
- document the allowed modes
- add checks that only allowed modes are accepted and raise `ValueError` for everything else
- add tests

Anything else?
